scalatest/src/main/scala/org/scalatest/prop/Randomizer.scala

```diff
@@ -138,8 +138,38 @@ class Randomizer(private[scalatest] val seed: Long) { thisRandomizer =>
     * @return A random Float in that range, and the next Randomizer to use.
     */
   def nextFloatBetween0And1: (Float, Randomizer) = {
-    val (i, r) = thisRandomizer.next(24)
-    (i / ((1 << 24).toFloat), r)
+    /*
+    0.0f is 32 zero bits. The sign is 0 for positive. The exponent of 0 is a special
+    case for 0.0f and for subnormal values. It means 0 if the mantissa is also 0.
+    1.0f is represented as 1.00 * (2 ** 0). The 1 to the left of the binary point
+    is implicit, so this means that the mantissa is all zeros again for 1.0. An
+    exponent of 0 is represented by 127 (0x7f). (To get the actual exponent from the biased
+    one, subtract 127.)
+
+    So except for 0.0f and 1.0f, which we'll need to handle specially, we can generate
+    a random 23 bits for the manitissa, and for the exponent a number between 1 and 126 (0x7e).
+    */
+
+    // The total number of possible images for Float between 0.0 and 1.0 is:
+    // (2 ** 23) * (126) + 2
+    // 2 to the power of 23 (0x800000) is how many different mantissas can be represented in 23 bits.
+    // 126 (or 0x7e) is how many exponents there are for numbers between 0.0f and 1.0f, exclusive.
+    // 2 is for 0.0f and 1.0f, which we must handle specially, because their mantissas must be all zeros.
+    // scala> 0x800000 * 0x7e + 2
+    // res14: Int = 1056964610
+    val (x, r) = chooseInt(0, 1056964610)
```

Comment:
Same comment as the other review: it's a bit-centric number, so I'd write this in hex.

Reply:
Will make it hex.